Modules/posixmodule.c

```diff
@@ -5088,384 +5088,358 @@ os__path_splitroot_impl(PyObject *module, path_t *path)
 }
 
 
-/*[clinic input]
-os._path_isdir
-
-    s: 'O'
+#define PY_IFREG  1 // Regular file
```

Comment:
Are these functions possibly reusable enough to put them in `fileutils.c` instead of temporarily defining them here?

Reply:
In principle they're reusable, but I haven't researched if there's a use for them in "fileutils.c" or the low-level startup code for interpreter initialization, getpath, etc.

The long-term goal, I think, is that once supported Windows versions (or at least the common supported versions) support `GetFileInformationByName()`, the implementation of `_path_[l]exists` and `_path_is` should switch to using `[L]STAT`. At that point, there won't be as much need to optimize the slow path. Using `[L]STAT` still provides the benefit of bypassing the overhead of a Python function call and the overhead of raising and ignoring `OSError` or `ValueError`. POSIX could benefit from these builtins as well. I guess this would be in Python 3.14 (2025-10-01), after Windows 10 is no longer supported.